text/0000-minimal-builders.md

```diff
@@ -0,0 +1,246 @@
+```toml
+[[build.buildpacks]]
+uri = "gcr.io/paketo-buildpacks/dotnet-core"
+```
+
+And pass the project descriptor to the `pack build` like:
+```
+pack build dotnet-with-minimal-builder \
+           --project-file project.toml
+           --builder gcr.io/paketo-buildpacks/builder:minimal-base
+```
+
+In fact, `pack` already supports supplying a `builder` key in a project file as
+of pack [v0.18.1](https://github.com/buildpacks/pack/releases/tag/v0.18.1).
+This will [soon be in the project descriptor
+spec](https://github.com/buildpacks/spec/issues/215) as well. Users can then
+set up a project.toml:
+```toml
+[build]
+builder = "gcr.io/paketo-buildpacks/builder:minimal-base"
+
+[[build.buildpacks]]
+uri = "gcr.io/paketo-buildpacks/dotnet-core"
+```
+
+and then run the build:
+```
+pack build dotnet-with-minimal-builder --project-file project.toml
+```
+
+## Rationale and Alternatives
+
+1. Do nothing. Assume that mature Paketo users will set up their own builders
+  and that this doesn't pose too much maintenance burden for them.
+   - Benefits:
+     - No work required from Paketo project.
+   - Drawbacks:
+     - All of the pain points described in the Motivation section of this RFC.
+
+1. Publish a Github Action, Concourse task, and other pre-made CI/CD tooling
+   templates that use `jam update builder` to keep users' custom builders up to
+   date.
+   - Benefits:
+     - Users can benefit from custom builders with diminished time investment
+   - Drawbacks:
+     - Paketo isn't a CI/CD tooling-focused project. This would expand the scope
+     of released artifacts for which the project is responsible.
+
+1. Make upstream contributions to pack that speed up builder pull times
+   - Benefits:
+     - Addresses the platform's problem of slow build times more directly. Will
+      benefit performance for users with _all_ builders.
+   - Drawbacks:
+     - Requires upstream contribution that may require an RFC. Optimizing image
+    pulls from a registry does not necessarily fall in scope of the Paketo
+    project currently. May require Paketo contributors to build up expertise
+    that is tangential to Paketo's main aims.
+
+<!--- Some other alternatives, deprioritized for now:
+- publish minimal builders, but check in their `builder.toml` files to the same
+  repos where the standard builders are. This increases discoverability of
+  buildpack compatibility. May also increase complexity of automations +
+  maintenance - we'd need to publish 2 github releases to the same repo. Github's support for this is poor.
+- instead of minimal builders, publish builders with curated sets of commonly-combined buildpacks (e.g. builders for each language family buildpack in isolation)
+--->
+
+## Implementation
+
+### Repositories
+Three new repositories should be created, for each of the new minimal builders:
+- paketo-buildpacks/minimal-full-builder
+- paketo-buildpacks/minimal-base-builder
+- paketo-buildpacks/minimal-tiny-builder
```

Comment:
I wonder if we should call these empty builders or something else instead of minimal (empty still isn't great... naming is hard!). To me, the word minimal in this context evokes minimal images like tiny and might not convey that these builders are devoid of buildpacks.

Reply:
I had the same thought. Maybe something like `buildpackless` or just `packless`.

Reply:
+1 empty was the first word that came to mind but probably isn't specific enough. If we can spare the characters, I'd prefer something more specific like `no-buildpacks` or `buildpackless`.